### ISSUE TITLE
[A11y] SemanticsFlag refactoring

### ISSUE THREAD

**Issue by hannah-hyj** [flutter/flutter#166101]
### Use case

The SemanticsFlag is running out 32 bits. 

### Proposal

Propose to refactor the semanticsFlags to a list of booleans

**Comment by jonahwilliams** [comment]
This might be a difficult change to make, as both dart:ui and all of the native semantics handling is passing around the `flags` as an integer. You might have an easier time adding `int secondaryFlags`, then we can start the counter again.

**Comment by hannah-hyj** [comment]
If we just add a new integer, it might running out again in the future.  Maybe it's better if we switch to use a list of booleans for good?

**Comment by jonahwilliams** [comment]
hopefully we stop adding flags at some point :).

Also we can use up to 53 bits even on JS

**Comment by hannah-hyj** [comment]
I'm on board with just adding a new integer like flag2.

As for 32 or 53, 
@chunhtai told me that android still has 32 bit devices and it will become a bottleneck.

**Comment by jonahwilliams** [comment]
That only impacts word size. Using a 64 bit int is fine on Android.

**Comment by hannah-hyj** [comment]
😮 I didn't know that. 

Good to know that we still have 20 more flag slots left.  In this case, I don't think we need to do anything for now.

**Comment by chunhtai** [comment]
How would adding an boolean list more difficult than adding a new int?

cc @yjbanov to confirm that web int has 64 bits

**Comment by jonahwilliams** [comment]
The web doesn't have 64 bit. I believe you have up to 53 bits on the web.

**Comment by yjbanov** [comment]
Unfortunately, the web only has 32 bits for the kinds of bitwise operations we need for flag and actions.

**Comment by yjbanov** [comment]
More on this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators#binary_bitwise_operators

**Comment by yjbanov** [comment]
But a class with a bunch of `bool` fields is probably fine too. We already use plain bools in `SemanticsProperties`. I doubt using the same representation in the engine will introduce any noticeable overhead.

**Comment by chunhtai** [comment]
bump this to p1 since this will likely block numerous work
